### PR TITLE
Change the JSON output format for the `kcp rc info` method

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- Adopt KCP CLI to the API changes introduced by #427 and #437

**Related issue(s)**
https://github.com/kyma-incubator/reconciler/issues/433

### CODE REVIEW

File: components/reconciler/makefile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 OAPI_FILE=./docs/external_api.yaml
-OAPI_REVISION=05b7b7a4d5d503972dcfb2a6d1397ee15cf6307f
+OAPI_REVISION=modify-reconciliation-info-endpoint
 
 CLIENT_FILE=./pkg/client_gen.go
 TYPES_FILE=./pkg/types_gen.go
 
 docs/external_api.yaml:
-	curl -o $(OAPI_FILE) --create-dirs https://raw.githubusercontent.com/kyma-incubator/reconciler/$(OAPI_REVISION)/openapi/external_api.yaml
+	curl -o $(OAPI_FILE) --create-dirs https://raw.githubusercontent.com/dbadura/reconciler/$(OAPI_REVISION)/openapi/external_api.yaml
```
Comment (pPrecel) [inline components/reconciler/makefile:9]: remember to revert these changes after merging [this PR](https://github.com/kyma-incubator/reconciler/pull/427) :)

File: tools/cli/pkg/command/reconciliation_info_test.go [hunk 2]
```diff
@@ -93,6 +95,41 @@ func TestReconciliationOperationInfoCommand_Run(t *testing.T) {
...
+		{
+			name: "reconciliation info: happy path - reconciliation found",
+			fields: fields{
+				ctx:    testCtx,
+				output: outputJSON,
+				provideMshipClient: func(url string, _ *http.Client) (mothership.ClientInterface, error) {
+
+					r, err := ioutil.ReadFile("testdata/reconciliationInfoResponse.json")
+					if err != nil {
+						fmt.Print(err)
```
Comment (dbadura) [inline tools/cli/pkg/command/reconciliation_info_test.go:107]: extract it to seperate method and add as first arg `testing.T` to abort test in case of error

File: tools/cli/pkg/command/reconciliation_info_test.go [hunk 3]
```diff
@@ -93,6 +95,41 @@ func TestReconciliationOperationInfoCommand_Run(t *testing.T) {
...
+					}
+					c, err := ioutil.ReadFile("testdata/configVersionResponse.json")
+					if err != nil {
+						fmt.Print(err)
+					}
+					m := msmock.NewMockClientInterface(ctrl)
```
Comment (dbadura) [inline tools/cli/pkg/command/reconciliation_info_test.go:113]: Do we check if mocks were called?

File: tools/cli/pkg/command/reconciliation_info_test.go [hunk 4]
```diff
@@ -93,6 +95,41 @@ func TestReconciliationOperationInfoCommand_Run(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "reconciliation info: happy path - reconciliation found",
```
Comment (dbadura) [inline tools/cli/pkg/command/reconciliation_info_test.go:99]: Do we check the output of this command or only error?

File: tools/cli/pkg/command/reconciliation_info.go [hunk 5]
```diff
@@ -140,8 +142,27 @@ func (cmd *ReconciliationOperationInfoCommand) Run() error {
 		return err
 	}
 
-	var result mothership.ReconcilationOperationsOKResponse
-	if err := json.NewDecoder(response.Body).Decode(&result); err != nil {
+	var result ReconcilerInfoResponses
```
Comment (dbadura) [inline tools/cli/pkg/command/reconciliation_info.go:145]: Split this code to 3 phases:
1 - init client
2 - get reconciliation info
3- get cluster config

This will help to avoid double closing the same response and not closing the first response by defer.

**Review by dbadura — CHANGES_REQUESTED** [review]

File: tools/cli/pkg/command/reconciliation_info_test.go (hunk 3 above)
Comment (kwiatekus) [reply tools/cli/pkg/command/reconciliation_info_test.go:104]: nope. but the logic printing the result to the output is not accessible outside that command to unit test.
That would require some heavy refactor

Comment (kwiatekus) [reply tools/cli/pkg/command/reconciliation_info_test.go:104]: yes we do. The mocking lib assures that via `.Times(x)`

File: tools/cli/pkg/command/reconciliation_info_test.go (hunk 4 above)
Comment (kwiatekus) [reply tools/cli/pkg/command/reconciliation_info_test.go:99]: nope. but the logic printing the result to the output is not accessible outside that command to unit test.
That would require some heavy refactor

File: tools/cli/pkg/command/reconciliation_info.go (hunk 5 above)
Comment (kwiatekus) [reply tools/cli/pkg/command/reconciliation_info.go]: done

File: tools/cli/pkg/command/reconciliation_info_test.go (hunk 3 above)
Comment (dbadura) [reply tools/cli/pkg/command/reconciliation_info_test.go:104]: I didn't notice the start of the test:
```
	ctrl := gomock.NewController(t)
	defer ctrl.Finish()
```

**Review by dbadura — APPROVED** [review]
Good work :)
